src/shared/dropdown-list/dropdown-list.tsx

```diff
@@ -172,7 +172,7 @@ export const DropdownList = <T, V>({
     };
 
     const checkListItemSelected = (item: T): boolean => {
-        return findIndex(selectedItems, item as any) > -1;
+        return indexOf<T>(selectedItems, item) > -1;
```

Comment:
it will be more reliable to use `valueExtractor` to compare items instead of relying on object reference. although I understand there's currently no requirement that value has to be unique

the below doesn't work, for example

```
<Form.MultiSelect
    label="This is the select field"
    options={[
        { value: "A", label: "Option A" },
        { value: "B", label: "Option B" },
        { value: "C", label: "Option C" },
        { value: "D", label: "Option D" },
    ]}
    selectedOptions={[{ value: "A", label: "Option A" }]}
    valueExtractor={(item) => item.value}
    listExtractor={(item) => item.label}
    displayValueExtractor={(item) => item.label}
/>
```

alternatively you could use `isMatches` as the comparator only when item is typeof object

Reply:
Ah my bad. i failed to check properly. Tested on the first few examples which did not have the `selectedOptions` passed. Will take your considerations into account